azure/src/test/java/com/microsoft/azure/management/TestTrafficManager.java

```diff
@@ -0,0 +1,324 @@
+        Assert.assertEquals(profile.azureEndpoints().size(), 1);
+        Assert.assertEquals(profile.nestedProfileEndpoints().size(), 1);
+        Assert.assertEquals(profile.externalEndpoints().size(), 2);
+
+        int c = 0;
+        for (TrafficManagerExternalEndpoint endpoint : profile.externalEndpoints().values()) {
+            Assert.assertEquals(endpoint.endpointType(), EndpointType.EXTERNAL);
+            if (endpoint.name().equalsIgnoreCase(externalEndpointName21)) {
+                Assert.assertEquals(endpoint.routingPriority(), 1);
+                Assert.assertEquals(endpoint.fqdn(), externalFqdn21);
+                Assert.assertNotNull(endpoint.monitorStatus());
+                Assert.assertEquals(endpoint.sourceTrafficLocation(), Region.US_EAST);
+                c++;
+            } else if (endpoint.name().equalsIgnoreCase(externalEndpointName22)) {
+                Assert.assertEquals(endpoint.routingPriority(), 2);
+                Assert.assertEquals(endpoint.fqdn(), externalFqdn22);
+                Assert.assertNotNull(endpoint.monitorStatus());
+                Assert.assertEquals(endpoint.sourceTrafficLocation(), Region.US_EAST2);
+                c++;
+            }
+        }
+        Assert.assertEquals(c, 2);
+
+        c = 0;
+        for (TrafficManagerAzureEndpoint endpoint : profile.azureEndpoints().values()) {
+            Assert.assertEquals(endpoint.endpointType(), EndpointType.AZURE);
+            if (endpoint.name().equalsIgnoreCase(azureEndpointName)) {
+                Assert.assertEquals(endpoint.routingPriority(), 3);
+                Assert.assertNotNull(endpoint.monitorStatus());
+                Assert.assertEquals(endpoint.targetAzureResourceId(), publicIpAddress.id());
+                Assert.assertEquals(endpoint.targetResourceType(), TargetAzureResourceType.PUBLICIP);
+                c++;
+            }
+        }
+        Assert.assertEquals(c, 1);
+
+        c = 0;
+        for (TrafficManagerNestedProfileEndpoint endpoint : profile.nestedProfileEndpoints().values()) {
+            Assert.assertEquals(endpoint.endpointType(), EndpointType.NESTEDPROFILE);
+            if (endpoint.name().equalsIgnoreCase(nestedProfileEndpointName)) {
+                Assert.assertEquals(endpoint.routingPriority(), 4);
+                Assert.assertNotNull(endpoint.monitorStatus());
+                Assert.assertEquals(endpoint.minChildEndpoints(), 1);
+                Assert.assertEquals(endpoint.nestedProfileId(), nestedProfile.id());
+                Assert.assertEquals(endpoint.sourceTrafficLocation(), Region.INDIA_CENTRAL);
+                c++;
+            }
+        }
+        Assert.assertEquals(c, 1);
+        return profile;
+    }
+
+    @Override
+    public TrafficManagerProfile updateResource(TrafficManagerProfile profile) throws Exception {
+        // Remove an endpoint, update two endpoints and add new one
+        //
+        profile.update()
+                .withTtl(600)
```

Comment:
i'm not sure if TTL is recognizable enough - perhaps spelling it out as withTimeToLive?